### DISCUSSION TITLE
Use nix to resolve gleam build/test/dev dependencies for local development

### DISCUSSION THREAD

**Discussion by oneness** [gleam-lang/gleam#4091] [Show and tell]
Hello folks, 
I have contributed to gleam few years back (pretty printing algo, gleam shell and couple of other mics type changes). I am thinking of starting to contribute again as long as my time allows.  As a long time nix user, I wrote a flake to get a terminal shell with all the gleam dependencies so I can run all the make targets (such as test, test-watch etc). I could contribute it to the repo if there is interest and consensus.  Having said that, nix is known for its complexity and I am always hesitant to introduce it to a code base where deterministic local dev dependencies is not huge pain point.  Nonetheless, here is the flake.nix with its entirety and it should work on linux, osx with various architecture. 

```
{
  description = "Gleam Development Shell";

  inputs = {
    flake-utils.url = "github:numtide/flake-utils";

    # You can change it based on what Gleam needs for development
    gleam_deps.url = "github:NixOS/nixpkgs/nixos-unstable";
  };

  outputs = { self, flake-utils, gleam_deps }@inputs :
    flake-utils.lib.eachDefaultSystem (system:
    let
      gleam_deps = inputs.gleam_deps.legacyPackages.${system};
    in
    {
      devShells.default = gleam_deps.mkShell {
        packages = [
          gleam_deps.cargo
          gleam_deps.clippy
          gleam_deps.erlang
          gleam_deps.nodejs
          gleam_deps.deno
          gleam_deps.rebar3
          gleam_deps.elixir
          gleam_deps.watchexec
        ];

        shellHook = ''
          echo "Welcome to Gleam Development Shell!"
          echo "Run:  'make help' to see available targets"
          
        '';
      };
    });
}
```

So, please chime in if you guys would like to have this in the main repo.

**Comment by lpil** [comment]
We do not support Nix in any way and will not accept any Nix related contributions. Nix produces more demand on Gleam than all other tools and environments combined, and so has been a hindrance to my work.

I'm sorry to be so negative, but debugging Nix caused bugs and responding to demands from Nix users (not you, you've been very polite and I am grateful!) has been frustrating and has worn me down.

**Reply by oneness** [reply]
@lpil I hear you.  Nix does introduce huge complexity.  So I am totally fine with this just being a template for those happen to use it. Feel free to delete/close or mark it as stale.

**Comment by oneness** [comment]
Closing this since the decision is clear from @lpil (No support for nix or any future nix related contributions)
